prospero-cli/src/main/java/org/wildfly/prospero/cli/RepositoryDefinition.java

```diff
@@ -17,32 +17,45 @@
 
 package org.wildfly.prospero.cli;
 
+import org.jboss.logging.Logger;
 import org.wildfly.channel.Repository;
 
+import java.io.File;
 import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URISyntaxException;
 import java.net.URL;
+import java.nio.file.Path;
+import java.nio.file.Paths;
 import java.util.ArrayList;
 import java.util.List;
 
 public class RepositoryDefinition {
 
+    private static final Logger logger = Logger.getLogger(RepositoryDefinition.class.getName());
+
     public static List<Repository> from(List<String> repos) throws ArgumentParsingException {
         ArrayList<Repository> repositories = new ArrayList<>(repos.size());
         for (int i = 0; i < repos.size(); i++) {
-            final String text = repos.get(i);
-            if(text.contains("::")) {
-                final String[] parts = text.split("::");
+            String repoName = repos.get(i);
+            if(repoName.contains("::")) {
+                final String[] parts = repoName.split("::");
                 if (parts.length != 2 || parts[0].isEmpty() || parts[1].isEmpty() || !isValidUrl(parts[1])) {
-                    throw CliMessages.MESSAGES.invalidRepositoryDefinition(text);
+                    throw CliMessages.MESSAGES.invalidRepositoryDefinition(repoName);
                 }
-
                 repositories.add(new Repository(parts[0], parts[1]));
             } else {
-                if (!isValidUrl(text)) {
-                    throw CliMessages.MESSAGES.invalidRepositoryDefinition(text);
+                if (!repoName.startsWith("http") && !repoName.isEmpty()) {
```

Comment:
should check if it starts with either `http://` or `https://`.

Reply:
if it starts with "https", it should be handled differently?

Reply:
Both http and https should be treated the same - as web URLs. The difference is that `http://foo.bar` is an URL `http/foo/bar` is a path